### PR TITLE
[Abandoned, author unable to close or comment] Clean up layout issues, handle large text sizes.

### LINKED ISSUE #303: UI Overlap on large text
there's some UI overlap of text overlapping the pager indicator when checking on a Nexus 5 emulator

it doesn't overlap on iPhone 8 but there is a margin issue, I'm not sure if that will block release

<details><summary>Android Nexus 5 (1080x1920 pixels) - click to expand</summary>

Screenshot_1585136324

Screenshot_1585136488

</details>

<details><summary>iPhone 8 simulator - click to expand</summary>

Simulator Screen Shot - iPhone 8 - 2020-03-25 at 12 45 39

</details>

### PR DESCRIPTION
**Pull request by gspencergoog**
## What does this PR accomplish?

This cleans up some of the existing overflow issues, and helps support large font sizes that users can set on iOS (and Android too, but there they don't get nearly as large).

I also made the main menu and cards scrollable if they overflow, with a gradient hint at the bottom if there is more to see.

Fixes #303

## Did you add any dependencies?

 - I bumped the min Dart version number to 2.3.0 from 2.1.0 to pick up many good Dart enhancements (list embedded conditionals, for one).

## How did you test the change?

I tested it in Android and iOS simulators.

Home screen:
Regular user text size:
Screen Shot 2020-03-25 at 4 53 46 PM

iOS largest accessibility text size:
Screen Shot 2020-03-25 at 4 54 52 PM

Myth-busters:
Regular size text:
Screen Shot 2020-03-25 at 5 06 11 PM

iOS largest accessibility text size:
Screen Shot 2020-03-25 at 5 06 51 PM

### CODE REVIEW

File: client/flutter/lib/carousel_page.dart [hunk 1]
```diff
@@ -59,9 +91,7 @@ class CarouselView extends StatelessWidget {
           ),
           Align(
             alignment: FractionalOffset.bottomCenter,
-            child: Container(
-                padding: EdgeInsets.only(bottom: 20),
-                child: pageViewIndicator(context)),
+            child: Container(padding: EdgeInsets.only(bottom: 20), child: pageViewIndicator(context)),
```
Comment (hspinks) [inline client/flutter/lib/carousel_page.dart]: is there a preferred code style on this from `flutter format`? prefer to standardize once and stick with it then have changes in a PR to review that aren't related to the overall purpose of the PR. not a blocker on this PR, just a general comment...

Comment (sstur) [reply client/flutter/lib/carousel_page.dart]: One style note on this, both usage of `EdgeInsets` above use `const` and use `10.0` but this uses the integer notation and no const. I don't know which one is preferable but we should probably be consistent.

**Comment by patniemeyer** [comment]
I think we should consider advanced accessibility options to be a separate topic from general adaptive text within the standard range and standard variation in screen sizes.  In general I think accommodating the full range of accessibility options may require some special work for special cases and I'm not sure that we'll be able to cover these in the first few releases.  That said, thank you for looking at this and I'll review what you did ASAP.

**Comment by patniemeyer** [comment]
I think this looks good.  A couple of things:  1) Wondering if we should center or offset the text when there is no image in the carousel?  This is more of a design decision so not a blocker.  2) Was the explicit scroll controller necessary for the `ScrollDownHint` (I haven't used that before).  Wondering if we could use `PrimaryScrollController.of(context)` here instead of creating one.  I think that may be necessary to get iOS to do the scroll to top on double tap gesture.

File: client/flutter/lib/carousel_page.dart (hunk 1 above)
Comment (gspencergoog) [reply client/flutter/lib/carousel_page.dart]: I looked for a `CONTRIBUTING.md` and in the Wiki to see, but guidance doesn't appear to exist, so I'm not sure what the coding conventions should be. My suggestion is to define it as the output of `flutter format -l 100 .` and leave it at that. Easily enforceable, eliminates all the time and effort talking about and fixing formatting, and it does a pretty good job, even if it's not perfect.

**Comment by gspencergoog** [comment]
The scroll controller is necessary, as the `ScrollDownHint` (a widget I added, not a Flutter widget) draws its hint on top of the scrollable, and so must be an ancestor of the Scrollable, which means that `PrimaryScrollController.of` won't be able to find the scrollable's controller. It uses the shared scroll controller to know when to display the hint.

I start-aligned the text when there is no icon.  It looks a little better (in my non-designer engineer opinion), so I kept it that way, but of course a designer may disagree. It's easy to change.

**Comment by patniemeyer** [comment]
@SamMousa I think this is ready to go but we need an approver with write permissions.

**Comment by SamMousa** [comment]
Will merge when at pc

File: client/flutter/lib/main.dart [hunk 2]
```diff
@@ -30,13 +30,14 @@ class _MyAppState extends State<MyApp> {
         S.delegate
       ],
       supportedLocales: S.delegate.supportedLocales,
-      theme: ThemeData(
+      theme: ThemeData.light().copyWith(
```
Comment (theswerd) [inline client/flutter/lib/main.dart]: I believe theme defaults to light and this is unnecessary

Comment (gspencergoog) [reply client/flutter/lib/main.dart]: It does, just being explicit. I'll revert this line.

File: client/flutter/lib/carousel_page.dart (hunk 1 above)
Comment (json469) [reply client/flutter/lib/carousel_page.dart]: If you would like to easily enforce but eliminate all the time and effort talking about formatting once-in-for-all 😆I would strongly recommend sticking to the [style guide for Flutter's own repo](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#formatting). For example: preferred maximum line length is 80.

Would be greatly beneficial to all new contributors even if we just link the above official Flutter repo's style guide in `CONTRIBUTING.md` so we are all in the same page. I spent some time looking for one before I started to code but couldn't find anything.

**Comment by advayDev1** [comment]
Now that the repo is opened up again, shall we investigate this in terms of the most recent design?

**Pull request closed without merging**